includes/class-amp-theme-support.php

```diff
@@ -1840,8 +1871,11 @@ public static function prepare_response( $response, $args = [] ) {
 			);
 		}
 
-		// Abort if the response was not HTML.
-		if ( 'text/html' !== substr( AMP_HTTP::get_response_content_type(), 0, 9 ) || '<' !== substr( ltrim( $response ), 0, 1 ) ) {
+		/*
+		 * Abort if the response was not HTML. To be post-processed as an AMP page, the output-buffered document must
+		 * have the HTML mime type and it must start <html> followed by <head> tag (with whitespace, doctype, and comments optionally interspersed).
+		 */
+		if ( 'text/html' !== substr( AMP_HTTP::get_response_content_type(), 0, 9 ) || ! preg_match( '#^(?:<!.*?>|\s+)*<html.*?>(?:<!.*?>|\s+)*<head.*?>#is', $response ) ) {
```

Comment:
The regexp here seems shaky. For example, an HTML fragment without a `<head>`, but containing a `<header>` element would still pass.